src/Elastic.Clients.Elasticsearch/_Shared/Serialization/DefaultSourceSerializer.cs

```diff
@@ -94,10 +100,13 @@ private static IReadOnlyCollection<JsonConverter> CreateDefaultBuiltInConverters
 
 	[UnconditionalSuppressMessage("AOT", "IL3050:Calling members annotated with 'RequiresDynamicCodeAttribute'", Justification = "Always using explicit TypeInfoResolver")]
 	[UnconditionalSuppressMessage("Trimming", "IL2026:Members annotated with 'RequiresUnreferencedCodeAttribute'", Justification = "Always using explicit TypeInfoResolver")]
-	private static void MutateOptions(JsonSerializerOptions options, IJsonTypeInfoResolver? typeInfoResolver, Action<JsonSerializerOptions>? configureOptions)
+	private static void MutateOptions(JsonSerializerOptions options, IJsonTypeInfoResolver[]? typeInfoResolvers, Action<JsonSerializerOptions>? configureOptions)
```

Comment:
Do we really need the multi-resolver overload?

I was under the impression that we can chain multiple `JsonTypeInfoResolver.Combine()` calls (given that the `System.Text.Json` API as well only accepts a single `IJsonTypeInfoResolver` in all places). This should give the user already all the required flexibility.

Reply:
Talked offline, reverted this!